### DISCUSSION TITLE
Troubleshooting

### DISCUSSION THREAD

**Discussion by versionDefect** [awawa-dev/HyperSerialEsp8266#24] [General]
I've got an ESP8266 (NodeMCU v3 CH340 https://www.amazon.com/gp/product/B09F8GCVC8/ref=ppx_yo_dt_b_asin_title_o03_s00?ie=UTF8&th=1)  with some WS2812B RGBNW and for some reason its just not doing anything. Logs don't say much https://pastebin.com/P0qma9DR. I know the LEDs work (tested them with ESPHome). 

I flashed it successfully. I've tried different USBs to connect from the Pi to the NodeMCU and connected them to Gpio 2 (D4). I am at a complete loss. It sees the device but I don't know what I'm doing wrong. 

Suggestions?

**Comment by awawa-dev** [comment]
Hi
Always provide full logs not just a part of it: don't see an issue here. Install v19beta2 from the release section and use ['esp32/esp8266 handshake'](https://github.com/awawa-dev/HyperHDR/wiki/HyperSerial) and provide logs again. Or you can compile the firmware online for 921600 already as fake ch340 is the most suspected reason. You can find manual here in the wiki.

**Comment by versionDefect** [comment]
> Hi
> 
> Always provide full logs not just a part of it: don't see an issue here. Install v19beta2 from the release section and use ['esp32/esp8266 handshake'](https://github.com/awawa-dev/HyperHDR/wiki/HyperSerial) and provide logs again. Or you can compile the firmware online for 921600 already as fake ch340 is the most suspected reason. You can find manual here in the wiki.

For one, Thanks for changing this to a discussion, Still pretty new to GitHub. Second, Damn, okay. I followed the wiki. Changed the baud rate, it compiled through GitHub like in the wiki. I downloaded the firmware and tried to put it on ESPHome flasher but it didn't work (it said it was not valid)? Am i doing something wrong?

**Comment by awawa-dev** [comment]
Did you extract the zip from Artifacts you built in Github Action and use firmware for LEDs type from this collection?

**Reply by versionDefect** [reply]
I swear I'm usually not this dumb. Reflashed and everything is working great. One last question for you. Going from Buad 921600 to 2000000 baud, is it much of a significant difference? 

I was previously using an Undervoltage Pi3 Gpio pin lol. I can already tell a significant difference.

**Reply by awawa-dev** [reply]
Glad it works now 👍 You can compare your HyperSerialEsp8266 statistics (FPS) with these I put on the main page here...it depends on the ESP UART and the LED number usually few percentage and for larger setup (~>=600) it's harder to maintain stable communication over USB serial connection at 1Mb. But there is no point to replace your esp8266 with another esp8266 with chip that supports 2Mb speeds, even if you don't get counterfeited chip still ESP32 with HyperSerialESP32 is ~10% more effective than esp8266 + has other features. So I suggest such upgrade in the future. Although all generic ESP32 should be supported, there are two recommended boards (including modern ESP32-S2 mini) on HyperSerialESP32 site ...so far we don't encounter counterfeits.

**Reply by awawa-dev** [reply]
How to read real ESP statistics: https://github.com/awawa-dev/HyperHDR/wiki/HyperSerial
